src/test/java/org/opentripplanner/middleware/controllers/api/ManageLegTraversalTest.java

```diff
@@ -18,66 +18,100 @@
 import static org.opentripplanner.middleware.triptracker.ManageLegTraversal.interpolatePoints;
 
 public class ManageLegTraversalTest {
 
-    private static MonitoredTrip monitoredTrip;
-    private static MonitoredTrip busStopJusticeCenterTrip;
-
-    private static final int SUBWAY_LEG_ID = 1;
-
-    private static final int FINAL_LEG_ID = 2;
+    private static Itinerary busStopToJusticeCenterItinerary;
 
     @BeforeAll
     public static void setUp() throws IOException {
-        monitoredTrip = JsonUtils.getPOJOFromJSON(
-            CommonTestUtils.getTestResourceAsString("controllers/api/monitored-trip-for-tracking.json"),
-            MonitoredTrip.class
-        );
-        busStopJusticeCenterTrip = JsonUtils.getPOJOFromJSON(
+        busStopToJusticeCenterItinerary = JsonUtils.getPOJOFromJSON(
             CommonTestUtils.getTestResourceAsString("controllers/api/bus-stop-justice-center-trip.json"),
-            MonitoredTrip.class
+            Itinerary.class
         );
     }
 
     @ParameterizedTest
     @MethodSource("createTrace")
     void canTrackTrip(String time, double lat, double lon, TripStatus expected) {
         TrackedJourney trackedJourney = new TrackedJourney();
-        trackedJourney.tripId = busStopJusticeCenterTrip.id;
         TrackingLocation trackingLocation = new TrackingLocation(time, lat, lon);
         trackedJourney.locations = List.of(trackingLocation);
-        TripStatus tripStatus = ManageTripTracking.getTripStatus(trackedJourney, busStopJusticeCenterTrip);
+        TripStatus tripStatus = ManageTripTracking.getTripStatus(trackedJourney, busStopToJusticeCenterItinerary);
         assertEquals(expected, tripStatus);
     }
 
     private static Stream<Arguments> createTrace() {
+        Date startTime = busStopToJusticeCenterItinerary.startTime;
+        List<ManageLegTraversal.Segment> segments = createSegmentsForLeg();
+        ManageLegTraversal.Segment before = segments.get(8);
+        ManageLegTraversal.Segment current = segments.get(10);
+        ManageLegTraversal.Segment after = segments.get(12);
         return Stream.of(
-            Arguments.of("2024-01-26T19:07:05Z", 33.95022, -83.9906, TripStatus.BEHIND_SCHEDULE),
-            Arguments.of("2024-01-26T19:07:08Z", 33.95022, -83.9906, TripStatus.ON_SCHEDULE),
-            Arguments.of("2024-01-26T19:07:50Z", 33.95022, -83.9906, TripStatus.AHEAD_OF_SCHEDULE),
-            Arguments.of("2024-01-26T19:07:43Z", 33.6408211227185, -84.4465224660685, TripStatus.DEVIATED),
-            Arguments.of("2024-01-26T19:12:25Z", 33.9517224, -83.9929082, TripStatus.NO_STATUS)
+            Arguments.of(
+                getDateTimeAsString(startTime, before.cumulativeTime),
+                current.start.lat,
+                current.start.lon,
+                TripStatus.BEHIND_SCHEDULE
+            ),
+            Arguments.of(
+                getDateTimeAsString(startTime, current.cumulativeTime),
+                current.start.lat,
```

Comment:
Can you add two cases please:
1. One with a tiny deviation on lat and lon that is still considered on schedule
2. Two with a tiny deviation regarding the time (i.e. `current.cumulativeTime +/- a few seconds) that are still considered "on-schedule".

Reply:
Addressed in https://github.com/ibi-group/otp-middleware/pull/214/commits/d4cc0bb523319ae7ee4eaa9d70446a8aa7850ee8